pkg/agent/controller/networkpolicy/audit_logging.go

```diff
@@ -133,10 +134,27 @@ func (l *AntreaPolicyLogger) updateLogKey(logMsg string, bufferLength time.Durat
 	return exists
 }
 
+func buildLogMsg(ob *logInfo) string {
+	var b strings.Builder
+	b.WriteString(ob.tableName)
+	b.WriteString(ob.npRef)
+	b.WriteString(ob.ruleName)
+	b.WriteString(ob.disposition)
+	b.WriteString(ob.ofPriority)
+	b.WriteString(ob.srcIP)
+	b.WriteString(ob.srcPort)
+	b.WriteString(ob.destIP)
+	b.WriteString(ob.destPort)
+	b.WriteString(ob.protocolStr)
+	b.WriteString(strconv.FormatUint(uint64(ob.pktLength), 10))
+	b.WriteString(ob.logLabel)
+	return b.String()
+}
+
 // LogDedupPacket logs information in ob based on disposition and duplication conditions.
 func (l *AntreaPolicyLogger) LogDedupPacket(ob *logInfo) {
 	// Deduplicate non-Allow packet log.
-	logMsg := fmt.Sprintf("%s %s %s %s %s %s %s %s %s %s %d %s", ob.tableName, ob.npRef, ob.ruleName, ob.disposition, ob.ofPriority, ob.srcIP, ob.srcPort, ob.destIP, ob.destPort, ob.protocolStr, ob.pktLength, ob.logLabel)
+	logMsg := buildLogMsg(ob)
 	if ob.disposition == openflow.DispositionToString[openflow.DispositionAllow] {
 		l.anpLogger.Printf(logMsg)
```

Comment:
I tried replacing it with `l.anpLogger.Print`. Surprisingly it is actually slower. My guess is that it is because of the variadic arguments (need to be converted to slice?).